### DISCUSSION TITLE
Apllying code in my own dataset

### DISCUSSION THREAD

**Discussion by nassera2014** [aub-mind/arabert#34] [Q&A]
Hi, Thank you for this great job,  i'm beginner in BERT, and i want to use it in topic modeling (extraction topics from **arabic** text), do you have an idea how can i do this using your code? thank you so much.

BR

**Comment by WissamAntoun** [comment]
Hey,

Try this project Top2Vec which uses pretrained models. You can use `aubmindlab/bert-base-arabertv01` with this library and it should work fine. https://github.com/ddangelov/Top2Vec

**Comment by nassera2014** [comment]
Thank you for your response , i tried your proposed code (Top2Vec), when i run the programme : 
`model = Top2Vec(documents=text_df[0][0], embedding_model="aubmindlab/bert-base-arabertv01")
print('num topics : ',model.get_num_topics())
print('topics :',model.get_topics())`

i get this error : 
) is an invalid embedding model.

Note that  i use Python 3.6

**Comment by WissamAntoun** [comment]
I looked into that problem, it seems like the model needs to first be trained using the https://www.sbert.net/docs/training/overview.html

Other than that I'm not sure how I can help
